File: androidApp/src/main/java/com/petros/efthymiou/dailypulse/android/PlatformInfo.kt
package com.petros.efthymiou.dailypulse.android

import com.petros.efthymiou.dailypulse.Platform

object PlatformInfo {
    private val platform = Platform()

    fun operatingSystem() = Pair("Operating System", "${platform.osName} ${platform.osVersion}")
    fun device() = Pair("Device", platform.deviceModel)
    fun density() = Pair("Density", "${platform.density} dots per inch (dpi)")

    fun platformItems():List<Pair<String,String>>{
        return listOf(
            operatingSystem(),
            device(),
            density(),
        )
    }
}

File: iosApp/iosApp/PlatformInfo.swift
//
//  PlatformInfo.swift
//  iosApp
//
//  Created by Danilo de Queiroz on 28/06/25.
//  Copyright © 2025 orgName. All rights reserved.
//
import shared
import Foundation

struct PlatformInfo {
    private let platform: Platform = Platform()

    func operatingSystem() -> (String, String) {
        return ("Operating System", "\(platform.osName) \(platform.osVersion)")
    }

    func device() -> (String, String) {
        return ("Device", platform.deviceModel)
    }

    func density() -> (String, String) {
        return ("Density", "Scale @\(platform.densityRound)x")
    }

    func platformItems() -> [(String, String)] {
        return [
            operatingSystem(),
            device(),
            density()
        ]
    }
}
